test/unit/dataLoaderUtil.test.js

```diff
@@ -0,0 +1,864 @@
+import sinon from 'sinon';
+import { expect } from 'chai';
+import dataLoaderUtil from '@dataLoaderUtil';
+import Actions from '@Actions';
+import Bib from '@Bib';
+import Search from '../../src/server/ApiRoutes/Search';
+import Hold from '@Hold';
+import routeMethods from '../../src/server/ApiRoutes/RouteMethods';
+
+describe('dataLoaderUtil', () => {
+  describe('server side call', () => {
+    describe('non-matching path', () => {
+      let axiosSpy;
+      let actionsSpy;
+      let sandbox;
+      let mockRouteMethods;
+      before(() => {
+        sandbox = sinon.createSandbox();
+        axiosSpy = sandbox.spy(axios, 'get');
+        mockRouteMethods = {
+          bib: sandbox.spy(),
+          search: sandbox.spy(),
+          hold: sandbox.spy(),
+        };
+        actionsSpy = [];
+        Object.keys(Actions).forEach((key) => {
+          if (typeof Actions[key] === 'function') {
+            actionsSpy.push(sandbox.spy(Actions, key));
+          }
+        });
+        const location = {
+          pathname: '',
+          search: '',
+        };
+        const req = true;
+        dataLoaderUtil.loadDataForRoutes(location, req, mockRouteMethods);
+      });
+      after(() => {
+        sandbox.restore();
+      });
+      it('should not call any function', () => {
+        const {
+          bib,
+          search,
+          hold,
+        } = mockRouteMethods;
+        expect(axiosSpy.getCalls()).to.have.lengthOf(0);
+        expect(bib.getCalls()).to.have.lengthOf(0);
+        expect(search.getCalls()).to.have.lengthOf(0);
+        expect(hold.getCalls()).to.have.lengthOf(0);
+        actionsSpy.forEach((spy) => {
+          expect(spy.getCalls()).to.have.lengthOf(0);
+        });
+      });
+    });
+    describe('bib path', () => {
+      let mockRouteMethods;
+      let bibCalls = [];
+      let bibResponse;
+      let actionsSpy;
+      let sandbox;
+      let axiosSpy;
+      let mockReq;
+      before(() => {
+        bibResponse = [];
+        mockRouteMethods = {
+          bib: (req, res) => {
+            bibCalls.push(req);
+            return res.json(bibResponse);
+          },
+        };
+        sandbox = sinon.createSandbox();
+        axiosSpy = sandbox.spy(axios, 'get');
+        actionsSpy = {};
+        Object.keys(Actions).forEach((key) => {
+          if (typeof Actions[key] === 'function') {
+            actionsSpy[key] = sandbox.spy(Actions, key);
+          }
+        });
+        const location = {
+          pathname: '/research/collections/shared-collection-catalog/bib/b000000',
+          search: '',
+        };
+        mockReq = [];
+        dataLoaderUtil.loadDataForRoutes(location, mockReq, mockRouteMethods);
+      });
+      after(() => {
+        sandbox.restore();
+      });
+      it('should not call axios', () => {
+        expect(axiosSpy.notCalled).to.equal(true);
+      });
+      it('should call bibSearch with request as argument', () => {
```

Comment:
I would say that it "should call mockRouteMethods.bib" rather than `bibSearch` since there's no reference to that function name in this file, and you wouldn't necessarily know if you didn't read the code, that it's `routeMethods.bib` that calls that function.

Reply:
And if you ever need to replace `bibSearch` with another function, you know that _this_ part will work as intended by calling whatever function `mockRouteMethods.bib`  is assigned to.